.github/workflows/run_integration_tests.yml

```diff
@@ -2,13 +2,406 @@ name: Run Integration Tests
 
 on:
   workflow_dispatch:
+    inputs:
+      run_id:
+        description: 'Run ID of the build workflow (all_solutions.yml) to use the agent and integration tests from. ID can be found in URL for run.'
+        required: true
+      solutions:
+        type: choice
+        description: Which test solutions (IntegrationTests, UnboundedIntegrationTests, or both) to run
+        options:
+        - IntegrationTests
+        - UnboundedIntegrationTests
+        - Both
+      integration-test-namespaces:
+        description: 'A comma-separated, single-quoted list of integration test namespaces to run (e.g. "''AgentFeatures'', ''BasicInstrumentation''").  If not specified, all integration tests will be run.'
+        required: true
+        default: 'ALL'
+      unbounded-test-namespaces:
+        description: 'A comma-separated, single-quoted list of unbounded integration test namespaces to run (e.g. "''MsSql'', ''Couchbase''").  If not specified, all unbounded tests will be run.'
+        required: true
+        default: 'ALL'
+      # iterations:
+      #   description: 'Number of times to execute each test, regardless of success or failure.  Defaults to 1.'
+      #   required: true
+      #   default: 1
+
+env:
+  DOTNET_NOLOGO: true
 
 jobs:
 
-  hello-world:
-    name: Placeholder job
+  get-external-artifacts:
+    name: Get and Publish Agent and Test Artifacts Locally
     runs-on: ubuntu-latest
     steps:
-    - name: Hello World
-      run: echo "Hello world!"
-      shell: bash
+      - name: Print input vars
+        run: |
+          echo "github.event.inputs.solutions = ${{ github.event.inputs.solutions }}"
+          echo "github.event.inputs.integration-test-namespaces = ${{ github.event.inputs.integration-test-namespaces }}"
+          echo "github.event.inputs.unbounded-test-namespaces = ${{ github.event.inputs.unbounded-test-namespaces }}"
+
+      - name: Download Agent Home Folders
+        uses: dawidd6/action-download-artifact@v2
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          workflow: all_solutions.yml
+          run_id: ${{ github.event.inputs.run_id }}
+          name: homefolders
+          path: ${{ github.workspace }}
+          repo: ${{ github.repository }}
+      
+      - name: Upload Agent Home Folders Locally
```

Comment:
If this step can download the home folders from another workflow/run, why do we need to re-upload them to this workflow/run?